### PR TITLE
Add GetLanguageFromRepositoryKey to Languages.cs

### PR DESCRIPTION
**Pull request by bigfluffycookie**
Fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/3600

### CODE REVIEW

File: src/Core.UnitTests/LanguageTests.cs [hunk 1]
```diff
@@ -100,5 +100,45 @@ public void Language_Equality()
             lang1b.Should().Be(lang1a, "Languages with the same ids should be equal");
             lang2.Should().NotBe(lang1a, "Languages with different ids should NOT be equal");
         }
+
+        [TestMethod]
```
Comment (bigfluffycookie) [inline src/Core.UnitTests/LanguageTests.cs:104]: I noticed that this wasn't tested before so I added it.

File: src/Core/Language.cs [hunk 2]
```diff
@@ -96,6 +101,19 @@ public static IEnumerable<Language> KnownLanguages
             }
         }
 
+        /// <summary>
+        /// Matches the repository name to its language object counterpart.
+        /// </summary>
+        private static readonly Dictionary<string, Language> repoKeyToLanguage = new Dictionary<string, Language>()
```
Comment (bigfluffycookie) [inline src/Core/Language.cs:107]: Are we using dictionaries for these kinds of things? To me it made sense but maybe I am wrong

Comment (duncanp-sonar) [reply src/Core/Language.cs:107]: Yep, makes sense.

File: src/Core/Language.cs [hunk 3]
```diff
@@ -56,6 +56,11 @@ public sealed class Language : IEquatable<Language>
         public static Language GetLanguageFromLanguageKey(string languageKey) =>
             KnownLanguages.FirstOrDefault(l => languageKey.Equals(l.ServerLanguage.Key, System.StringComparison.OrdinalIgnoreCase));
 
+        /// <summary>
+        /// Tries to get the language for the specified repository key.
+        /// </summary>
+        public static bool GetLanguageFromRepositoryKey(string repoKey, out Language language) => repoKeyToLanguage.TryGetValue(repoKey, out language);
```
Comment (duncanp-sonar) [inline src/Core/Language.cs:62]: It seems a bit odd that the `Get` methods have similar names (`GetLanguageFromLanguageKey`, `GetLanguageFromRepositoryKey`), but have different signatures.

Your new method is really a `TryGet`. Does it need to be?
If it does, the test needs to check the return value as well as the `out` value.

Comment (bigfluffycookie) [reply src/Core/Language.cs]: It does not need to be, not sure why I have made it that way

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**